### ISSUE TITLE
[RFC] yoshi-flow-editor CLI extension

### ISSUE THREAD

**Issue by yavorsky** [wix/yoshi#2520]
# 🙋 Feature Request

This is a RFC for `yoshi-flow-editor` CLI command tool extension.

Currently we support few basic commands:
- `start`: starts a development server with local overrides
- `build`: builds production bundles
- `test`: start unit and e2e tests with pre-configured jest under the hood
- `lint`:  lint check

All commands are working pretty good on the current stage. But it could be extended with bigger functionality.

### `test`
It starts a unit and e2e tests. e2e tests we have just for Settings, since Widget is being tested via `sled`.
There are few issues:
- Widget for Editor part is not being tested at all. Sled is testing it only in viewer environment.
- Settings is being tested w/o scope also: just serving an app on localhost and providing some e2e checks.
- This command is not starting sled tests. So usually we should have `yoshi-flow-editor test && sled`. It's not good practice in general, since we these commands is not running in parallel and and with `|` operator we will mix/lose some logs.

There're some ideas how we can improve the flow in general
- experiment with running sled against editor environment. We can already override editorScriptUrl: https://github.com/wix-private/sled/pull/541 and since we can already use query params to override everything we need to editor part, maybe few changes to the sled side will help is.
- Proxy sled command from the yoshi CLI and run it when we are running `test`. We can also support commands like `test:unit`, `test:sled`, so users can run some specific tests locally.

### `add`
A new command that should add new things to the current project.

Currently, we are bootstrapping almost everything needed for development and release with `create-yoshi-app` for editor flow. It's Dev Center registration for widgets, pages, platform Components with `viewerScript`, `editorScript` urls, configuring sentry, fedops, basic experiments bootstrap.

But on the same time it's really hard to add new component. You need to provide all the steps manually spending much more time than just generate a new project.

With `yoshi-flow-editor add` command we can open a wizard with questions to:
- add `Page` | `Widget`. In this case we'll provide the whole flow to correctly inject it in our project: Register on Dev Center, update `fedops.json`, create a correct `.component.json` config with component id, create a sled test for it.
- add `editor.app.ts` | `viewer.app.ts` (editorScript and viewerScript entry-points). In this case we'll just bootstrap it with some simple code `initAppForPage` or `editorReady`. It will be really useful if we want to remove this files from the `template`.
- add `sentry` - in case of changing the sentry team or adding a sentry to project that didn't configure a sentry, so correct `sentry` config will be updated in `.application.json`
- add `experiment`, so correct petry directory will be created for current scope and `scope` field will be updated in `.application.json` and experiments for this scope will be available via `flowAPI.getExperiments()`.

I would really love to hear more ideas about commands we can support, current solutions modifications and general idea overview.

It's something that we really need and a list of proposals is sorted by priority.

### `open`
It's some helper command that will open different useful urls we could need during development.
- `viewer`, `editor` - just opens it with correct overrides, the same we are doing for `start`.
- `sentry` - opens current configured sentry dashboard.
- `fedops` - opens grafana for current project
- `experiments` - opens guineapig website with current experiment
- `docs` - open editor flow documentation website
- More ?

It's low priority command, but would be really nice to have it.

**Comment by ronnyrin** [comment]
### `Tests`
> It starts a unit and e2e tests. e2e tests we have just for Settings, since Widget is being tested via sled.

Why not adding e2e tests to cover the editorMode without sled? And if we are already in the tests section, I think that all the tests (beside sled) in this flow are wrong as you want to check integration and not the worker and component separately. 

Note that `test` should run `perfer` as well.

### `Add`
> add Page | Widget

I don't see the value of it. First, you should strive to have one widget in one artifact. If you want to add a new page/widget, use `create-yoshi-app` in a new artifact. Second, personally i think that it's super risky to expose API that changes configuration in dev center behind the scenes. Every change can cause a real disaster in production and I think the dev should be super aware of what he is doing and why.

> sentry

sounds like an edge case that we shouldn't waste time on it.

### `Open`
In general, I don't think it's necessary.
> viewer, editor - just opens it with correct overrides, the same we are doing for start.

How is it going to serve the current statics? do you upload the statics like sled?

> fedops - opens grafana for current project

we have two dashboards (one for ooi and one for editor) so it will be too much to open both.
> experiments - opens guineapig website with current experiment

wdym? which experiment exactly?

**Comment by ranyitz** [comment]
Thanks @ronnyrin! It looks to me that these are 3 different issues. I believe that by splitting those we'll be able to communicate on each subject properly.
